synapse/rest/client/models.py

```diff
@@ -36,18 +36,20 @@ class Config:
     type: Optional[StrictStr] = None
 
 
-class ThreepidRequestTokenBody(RequestBodyModel):
-    if TYPE_CHECKING:
-        client_secret: StrictStr
-    else:
-        # See also assert_valid_client_secret()
-        client_secret: constr(
-            regex="[0-9a-zA-Z.=_-]",  # noqa: F722
-            min_length=0,
-            max_length=255,
-            strict=True,
-        )
+if TYPE_CHECKING:
+    ClientSecretType = StrictStr
+else:
+    # See also assert_valid_client_secret()
+    ClientSecretType = constr(
+        regex="[0-9a-zA-Z.=_-]",  # noqa: F722
+        min_length=0,
```

Comment:
'must not be empty' suggests min_length=1, but that's only a tiny quibble https://spec.matrix.org/v1.3/client-server-api/#post_matrixclientv3accountpasswordemailrequesttoken

Reply:
Oh good spot, thank you!